rero/rero-ils
# item: field call number is now optional

The item call number is now an optional field in both the
item json schema and item editor. This field is not populated
if not given by the librarian.

The item barcode remains a required field in the item json schema,
on the other side, it is an optional field in the item editor.

If a librarian decides not to add an item barcode, the system
will automatically populate the barcode with the following:
f-current timestamp.

* Adapts schema, tests and files correspondingly.

Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>
Co-Authored-by: Aly Badr<aly.badr@rero.ch>

## Why are you opening this PR?

https://tree.taiga.io/project/rero21-reroils/us/1328?milestone=263424

## How to test?

`bootstrap` + `setup`
in the UI, you should be able to add items without barcodes

## Code review check list
